.github/workflows/lint-test.yml

```diff
@@ -1,53 +1,93 @@
-name: Lint and Test
+name: Build, Lint, and Test
 
 on:
   push:
     branches: [main]
   pull_request:
 
 jobs:
-  lint-test:
-    name: Lint and Test
-    runs-on: ubuntu-20.04
+  prepare:
```

Comment:
Since we run `lint` for different Node.js versions, it may still be a good idea to keep the prepare workflow. Otherwise each `lint` job would be overwriting the same cache (as the Node.js version is not included in the cache key with `actions/setup-node`).